From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Varun Gupta <varun.gupta@intel.com>
Date: Fri, 10 Jul 2026 20:37:13 +0530
Subject: [PATCH] On v7.0+, zone_device_page_init() takes 3 args (page, pgmap,
 order). On v6.6 it takes only the page.

v2: Adding back this change post THP migration patch, due to dependency

Signed-off-by: Varun Gupta <varun.gupta@intel.com>
Signed-off-by: Pravalika Gurram <pravalika.gurram@intel.com>
---
 drivers/gpu/drm/drm_pagemap.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/drm_pagemap.c b/drivers/gpu/drm/drm_pagemap.c
index 35bfd48..c5cd39d 100644
--- a/drivers/gpu/drm/drm_pagemap.c
+++ b/drivers/gpu/drm/drm_pagemap.c
@@ -212,7 +212,11 @@ static void drm_pagemap_get_devmem_page(struct page *page,
 {
 #ifdef BPM_ZONE_DEVICE_FOLIO_INIT_NOT_PRESENT
 	page->zone_device_data = drm_pagemap_zdd_get(zdd);
+#ifdef BPM_ZONE_DEVICE_PAGE_INIT_3ARGS_NOT_PRESENT
+	zone_device_page_init(page);
+#else
 	zone_device_page_init(page, page_pgmap(page), 0);
+#endif
 #else
 	zone_device_folio_init((struct folio *)page, zdd->dpagemap->pagemap,
 			       order);
-- 
2.43.0

